### DISCUSSION TITLE
sharable config preset reference in a branch

### DISCUSSION THREAD

**Discussion by electriquo** [renovatebot/renovate#23921] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

there is ability to specify [config presets](https://docs.renovatebot.com/config-presets/#github) to use. from time to time, changes need to be introduced into a sharable config, which is done using a git branch and corresponding pull-request.

yet, renovate supports only a git tag but not a git branch. using a git tag is not enough, since a git tag is derived only after the side branch is merged into the main branch.

thus, when introducing a change to the sharable configuration, how can a git branch be specified in renovate config in order to test the change of the sharable configuration which present in a a side branch?

### What you tried so far.

review https://docs.renovatebot.com/config-presets/#github but there is no way to specify a git branch

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
The code which fetches it is here: https://github.com/renovatebot/renovate/blob/main/lib/config/presets/github/index.ts

It might be possible to substitute a ref (branch) for a tag and it simply works, otherwise you might find a way to make it work via a simple PR to the Renovate codebase
